### PR TITLE
Adds NodeEngine & Yarn buildpacks as optional

### LINKED ISSUE #651: Add Node into the buildpack order group
## What happened?

It is not uncommon to have a front-end and back-end app packaged together. The backend running on Java, of course :) and the front-end being HTML/Javascript. In this case, there are Maven/Gradle plugins that can be used to drive the build and also build the front-end assets and materials.

Right now, the same Maven/Gradle plugins would need to also install Node.js/NPM to work. We can add the `paketo-buildpacks/node-engine` buildpack to the order group and conditionally require it (probably from libbs) if one of the build tools requires Node.js/NPM.

This proposal would be to **only** install Node/NPM. Running commands would be up to Maven/Gradle. This is similar to how the Paketo Ruby buildpack installs Node.js/NPM and to enable Rails to build assets.

### PR DESCRIPTION
**Pull request by pivotal-david-osullivan**
Fixes #651 

## Use Cases
See paketo-buildpacks/gradle#216 and paketo-buildpacks/maven#239

## Checklist

* [ ] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by TisVictress** [comment]
Bump 🙂
Please let me know how I can help get this merged.

File: buildpack.toml [hunk 1]
```diff
@@ -55,6 +55,16 @@ api = "0.7"
...
+  [[order.group]]
+    id = "paketo-buildpacks/yarn"
+    optional = true
+    version = "1.0.0"
+
+  [[order.group]]
+    id = "paketo-buildpacks/node-engine"
+    optional = true
+    version = "1.1.0"
+
```
Comment (dmikusa) [inline buildpack.toml:67]: I think these should com after the JVM provider buildpack. Technically closure tools and leinigen are build related as well and could eventually support integration with Node also.

**Review by dmikusa — COMMENTED** [review]
A couple questions:

- how do we feel these Buildpacks will work together in terms of SBOM generation. I think it should be OK did you test it though?

- what are your thoughts on version/labels? Is this a minor change enhancement? Or major change? I don't think it's technically breaking but we do have a major change coming up. Should we wait to include it in the major change?

File: buildpack.toml (hunk 1 above)
Comment (pivotal-david-osullivan) [reply buildpack.toml]: That's a good point! 👍

**Comment by pivotal-david-osullivan** [comment]
> how do we feel these Buildpacks will work together in terms of SBOM generation. I think it should be OK did you test it though?

Keeping in mind the new buildpacks are build-only, I exported and took a look at the built-time SBOM for Gradle. This seems to contain all the npm SBOM info:
 {
      "bom-ref": "pkg:npm/%40rollup/plugin-node-resolve@11.2.1?package-id=4865bf6003def421",
      "cpe": "cpe:2.3:a:\\@rollup\\/plugin-node-resolve:\\@rollup\\/plugin-node-resolve:11.2.1:*:*:*:*:*:*:*",
      "licenses": [
        {
          "license": {
            "id": "MIT"
          }
        }
      ],
      "name": "@rollup/plugin-node-resolve",
      "properties": [
        {
          "name": "syft:package:foundBy",
          "value": "javascript-lock-cataloger"
        },
```

> what are your thoughts on version/labels? Is this a minor change enhancement? Or major change? I don't think it's technically breaking but we do have a major change coming up. Should we wait to include it in the major change?

We previously added the JAttach & JavaMemoryAssistant buildpacks with a minor bump & enhancement. Given that and the fact it shouldn't be breaking as you say, I think this should be okay here too.

**Review by dmikusa — APPROVED** [review]

**Pull request merged**